app/controllers/carts_controller.rb

```diff
@@ -1,24 +1,41 @@
 class CartsController < ApplicationController
+  before_action :find_current_cart
+  before_action :find_cart_products, only: :index
+  before_action :update_price, only: :index
+
   def index
```

Comment:
You already defined a logged in filter as I remember, you can just use that one.